web/src/app/pages/edit-survey/edit-survey.component.ts

```diff
@@ -61,18 +61,18 @@ export class EditSurveyComponent implements OnInit {
 
   ngOnInit(): void {
     this.subscription.add(
-      this.navigationService.getSurveyId$().subscribe(surveyId => {
+      this.navigationService.getSurveyId$().subscribe(async surveyId => {
         if (surveyId) {
           this.surveyId = surveyId;
           this.surveyService.activateSurvey(surveyId);
+          await this.draftSurveyService.init(surveyId);
+          this.draftSurveyService
+            .getTempSurvey$()
```

Comment:
Nit: Please remove "temp" prefix here and throughout. Temp makes it sound like it's a scrap copy for an interim operation. Unless it's ambiguous we shouldn't need to prefix with "draft" either, since the method is already scoped to the DraftSurveyService.